### ISSUE TITLE
TypeScript: @types/express version ambiguity breaks build

### ISSUE THREAD

**Issue by montasaurus** [firebase/firebase-functions#685]
<!-- DO NOT DELETE
validate_template=true
template_path=.github/ISSUE_TEMPLATE/---report-a-bug.md
-->

### Related issues
https://github.com/firebase/firebase-functions/issues/610
<!-- Are there any related firebase-functions issues that you found on this topic before deciding to open a new issue? Please link them here-->

### [REQUIRED] Version info

<!-- What versions of the following libraries are you using? Note that your issue may already
be fixed in the latest versions. -->

**node:** v13.12.0

<!-- Run node --version and print the output here-->

**firebase-functions:** 3.6.1

**firebase-tools:** 8.2.0

<!-- Run firebase --version and print the output here-->

**firebase-admin:** ^8.11.0

### [REQUIRED] Test case
```
import * as functions from "firebase-functions";
functions.https.onCall((data, context) => {}); // error TS2707: Generic type 'Request<P>' requires between 0 and 1 type arguments.
```
<!-- Provide a minimal, complete, and verifiable example (http://stackoverflow.com/help/mcve) -->

### [REQUIRED] Steps to reproduce
Upgrade the `firebase-functions` package to 3.6.1 with `@types/express` already installed and resolved to version 4.17.3 (which matches the spec for this package). Run `tsc` and watch the build fail.
<!-- Provide the steps needed to reproduce the issue given the above test case. -->

### [REQUIRED] Expected behavior
The build should succeed.
<!-- What is the expected behavior? -->

### [REQUIRED] Actual behavior
The build fails because the type signatures have changed in `@types/express` since the spec version in `package.json`. Breaking change here: https://github.com/DefinitelyTyped/DefinitelyTyped/commit/bb2673643a8292a565378ffe73b01c37aa7e1194

```
- error TS2707: Generic type 'Request<P>' requires between 0 and 1 type arguments.

64         onCall: (handler: (data: any, context: https.CallableContext) => any) => import("./cloud-functions").TriggerAnnotated & ((req: express.Request<import("express-serve-static-core").ParamsDictionary, any, any, import("express-serve-static-core").Query>, resp: express.Response<any>) => void) & import("./cloud-functions").Runnable<any>;
                                                                                                                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/firebase-functions/lib/function-builder.d.ts:64:252 - error TS2694: Namespace '"/Users/adam/code/functions/node_modules/@types/express-serve-static-core/index"' has no exported member 'Query'.

64         onCall: (handler: (data: any, context: https.CallableContext) => any) => import("./cloud-functions").TriggerAnnotated & ((req: express.Request<import("express-serve-static-core").ParamsDictionary, any, any, import("express-serve-static-core").Query>, resp: express.Response<any>) => void) & import("./cloud-functions").Runnable<any>;
                                                                                                                                                                                                                                                              ~~~~~
```
The DefinitelyTyped bindings don't usually respect SemVer, so they should probably be pinned to a specific version in `package.json` or locked with `package-lock.json`.
<!-- Please copy and paste any error logs from https://console.firebase.google.com/project/_/functions/logs.
     If you're experiencing a deployment issue, please copy and paste the entirety of firebase-debug.log -->

### Were you able to successfully deploy your functions?
No
<!-- When you ran `firebase deploy`, did you see any error messages? -->

**Comment by laurenzlong** [comment]
This has now been fixed in 3.6.2

**Comment by LFDM** [comment]
In case someone else stumbles across this:
The fix above only partially fixes the issues with the express types. The culprit lies deeper: `@types/express-serve-static-core` doesn't have a fixed version within `@types/express` either, so it can lag behind on your system - this is what happened for me.

For now I just manually added the latest version of `@types/express-serve-static-core` to my own `package.json` (4.17.8), which works fine.
At the end of the day this needs to be fixed in `@types/express`.
